sdk/src/main/java/com/supenta/flitchio/sdk/BroadcastReceiverWithCallback.java

```diff
@@ -0,0 +1,44 @@
+package com.supenta.flitchio.sdk;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.IntentFilter;
+import android.support.annotation.NonNull;
+
+/**
+ * Utility class for an easier definition of {@link BroadcastReceiver}s.
+ *
+ * @param <T> An interface for callbacks.
+ */
+abstract class BroadcastReceiverWithCallback<T> extends BroadcastReceiver {
+    private final IntentFilter intentFilter;
+    private T callback;
+
+    protected BroadcastReceiverWithCallback(String... actionsToWatch) {
+        intentFilter = new IntentFilter();
+        for (String action : actionsToWatch) {
+            intentFilter.addAction(action);
+        }
+    }
+
+    public void start(Context context, @NonNull T callback) {
```

Comment:
I like this class, consider adding '@NonNull' to the context as well thought because 'start' and 'stop' will throw an NPE.